### PR TITLE
Retry kube requests on 500s.

### PR DESCRIPTION
**Pull request by spxtr**

### CODE REVIEW

File: prow/kube/client.go [hunk 1]
```diff
@@ -104,8 +104,10 @@ func (c *Client) requestRetry(r *request) ([]byte, error) {
 	backoff := retryDelay
 	for retries := 0; retries < maxRetries; retries++ {
 		resp, err = c.doRequest(r.method, r.path, r.query, r.requestBody)
-		if err == nil {
+		if err == nil && resp.StatusCode < 500 {
```
Comment (rmmh) [inline prow/kube/client.go]: maybe nest the resp.StatusCode < 500 insidE?

    if err == nil {
        if resp.StatusCode < 500 {
            break
        }
        resp.Body.Close()
    }

Comment (kargakis) [inline prow/kube/client.go]: Curious why isn't this an error from doRequest?

Comment (spxtr) [inline prow/kube/client.go]: `doRequest` in a wrapper around an `http.Client.Do`, which doesn't return an error on 500s.

Comment (spxtr) [inline prow/kube/client.go]: Done.

**Comment by kargakis** [comment]
Should we bump all affected components for this change?

**Comment by spxtr** [comment]
Yes, #3452

**Pull request merged**